src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.StandardProperties.cs

```diff
@@ -164,7 +163,44 @@ public int KeySize
         {
             get
             {
-                int keySize = _keyHandle.GetPropertyAsDword(KeyPropertyName.Length, CngPropertyOptions.None);
+                int keySize = 0;
+                int cbResult;
+
+                unsafe
+                {
+                    // Attempt to use PublicKeyLength first as it returns the correct value for ECC keys
+                    ErrorCode errorCode = Interop.NCrypt.NCryptGetProperty(
+                        _keyHandle,
+                        KeyPropertyName.PublicKeyLength,
+                        &keySize,
```

Comment:
I thought about this originally, but the existing patterns (e.g. GetPropertyAsDword) return 0 if not found, or throw exception. Since we have to detect gracefully in the first case, I extended the existing code instead of creating a new pattern. However I just added a NCryptGetIntProperty that removes some of the burden but keeps the errorcode return value to detect error case